pants.toml

```diff
@@ -2,15 +2,11 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 [GLOBAL]
-pants_version = "1.30.0"
-v1 =  false  # Turn off the v1 execution engine.
-v2 = true  # Enable the v2 execution engine.
+pants_version = "2.0.0.dev4"
 pantsd = true  # Enable the Pants daemon for better performance.
 
-backend_packages = []  # Deregister all v1 backends.
-
-# List v2 backends here.
-backend_packages2.add = [
+# List backends here.
+backend_packages.add = [
   'pants.backend.awslambda.python',
   'pants.backend.codegen.protobuf.python',
```

Comment:
Commenting out the `pants.backend.codegen.protobuf.python` backend will get you a bit further (although those target types will fail to load, and you'll need to remove the `protoc` config section). I'm investigating why here: https://github.com/pantsbuild/pants/issues/10425

As reported there, this issue was introduced sometime after `dev2`: you might consider adjusting the pants version to `2.0.0.dev2` if you'd like to make further progress.

Reply:
Thanks, `dev2` worked! Is `backend_packages.add = [...]` or  just `backend_packages = [...]` the preferred way to go? Both return the same result for `./pants list ::` and `./pants test ::`, but `backend_packages = [...]` (w/o add) is documented in https://www.pantsbuild.org/v2.0/docs/how-to-upgrade-pants-2-0 .

Reply:
Technically, both work the same right now. We don't have any default backends activated and we're unlikely to. We decided to have it so that `backend.project_info` and `backend.pants_info` are always activated no matter which, so that you always have goals like `dependencies` which are important to debugging things.

It wasn't always this way, which is why we're being inconsistent. `.add` is safer, but not currently necessary.

Reply:
Thanks for the explanation, I'll keep the `.add` then.